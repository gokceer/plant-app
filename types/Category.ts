export interface CategoryImage {
  id: number;
  name: string;
  alternativeText: string | null;
  caption: string | null;
  width: number;
  height: number;
  formats: any;
  hash: string;
  ext: string;
  mime: string;
  size: number;
  url: string;
  previewUrl: string | null;
  provider: string;
  provider_metadata: any;
  createdAt: string;
  updatedAt: string;
}

export interface Category {
  id: number;
  name: string;
  createdAt: string;
  updatedAt: string;
  publishedAt: string;
  title: string;
  rank: number;
  image: CategoryImage;
}
