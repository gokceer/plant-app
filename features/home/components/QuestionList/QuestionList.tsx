import { Question } from "@/types/Question";
import React from "react";
import { ScrollView, StyleSheet, Text, View } from "react-native";
import QuestionCard from "../QuestionCard/QuestionCard";

const QuestionList = ({ questions }: { questions: Question[] }) => {
  return (
    <View>
      <Text style={styles.title}>Get Started</Text>
      <ScrollView
        horizontal
        showsHorizontalScrollIndicator={false}
        bounces={false}
        style={styles.container}
      >
        {questions.map((question, index) => (
          <QuestionCard
            key={index}
            image={question.image_uri}
            title={question.title}
          />
        ))}
      </ScrollView>
    </View>
  );
};

export default QuestionList;

const styles = StyleSheet.create({
  container: {
    marginLeft: 24,
    marginBottom: 24,
  },
  title: {
    fontSize: 15,
    fontFamily: "RubikMedium",
    color: "#13231B",
    marginBottom: 16,
    marginLeft: 24,
  },
});
