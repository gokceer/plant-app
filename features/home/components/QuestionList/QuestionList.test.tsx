import { Question } from "@/types/Question";
import { render } from "@testing-library/react-native";
import React from "react";
import QuestionList from "./QuestionList";

jest.mock("expo-blur", () => ({
  BlurView: jest.fn(({ children }) => children),
}));

const sampleQuestions: Question[] = [
  {
    id: 1,
    title: "How to water a cactus?",
    image_uri: "https://example.com/cactus.jpg",
    subtitle: "",
    uri: "",
    order: 1,
  },
  {
    id: 2,
    title: "Best sunlight for ferns",
    image_uri: "https://example.com/ferns.jpg",
    subtitle: "",
    uri: "",
    order: 2,
  },
];

describe("QuestionList", () => {
  it("renders correctly and matches snapshot", () => {
    const { toJSON } = render(<QuestionList questions={sampleQuestions} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders empty list gracefully", () => {
    const { getByText, queryByText } = render(<QuestionList questions={[]} />);
    expect(getByText("Get Started")).toBeTruthy();
    expect(queryByText("How to water a cactus?")).toBeNull();
  });
});
