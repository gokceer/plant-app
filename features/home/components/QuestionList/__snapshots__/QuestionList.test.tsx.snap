// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionList renders correctly and matches snapshot 1`] = `
<View>
  <Text
    style={
      {
        "color": "#13231B",
        "fontFamily": "RubikMedium",
        "fontSize": 15,
        "marginBottom": 16,
        "marginLeft": 24,
      }
    }
  >
    Get Started
  </Text>
  <RCTScrollView
    bounces={false}
    horizontal={true}
    showsHorizontalScrollIndicator={false}
    style={
      {
        "marginBottom": 24,
        "marginLeft": 24,
      }
    }
  >
    <View>
      <View
        style={
          {
            "borderRadius": 12,
            "height": 164,
            "marginRight": 10,
            "width": 240,
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            {
              "borderRadius": 12,
              "height": 164,
              "overflow": "hidden",
              "width": 240,
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://example.com/cactus.jpg",
              }
            }
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                {
                  "height": 164,
                  "width": 240,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              {
                "borderBottomLeftRadius": 12,
                "borderBottomRightRadius": 12,
                "flex": 1,
                "overflow": "hidden",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontFamily": "RubikRegular",
                  "fontSize": 15,
                  "paddingHorizontal": 14,
                  "paddingTop": 10,
                }
              }
            >
              How to water a cactus?
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "borderRadius": 12,
            "height": 164,
            "marginRight": 10,
            "width": 240,
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            {
              "borderRadius": 12,
              "height": 164,
              "overflow": "hidden",
              "width": 240,
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://example.com/ferns.jpg",
              }
            }
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                {
                  "height": 164,
                  "width": 240,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              {
                "borderBottomLeftRadius": 12,
                "borderBottomRightRadius": 12,
                "flex": 1,
                "overflow": "hidden",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontFamily": "RubikRegular",
                  "fontSize": 15,
                  "paddingHorizontal": 14,
                  "paddingTop": 10,
                }
              }
            >
              Best sunlight for ferns
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
